clang/lib/Sema/SemaDecl.cpp

```diff
@@ -3215,8 +3215,12 @@ static void adjustDeclContextForDeclaratorDecl(DeclaratorDecl *NewD,
 template <typename AttributeType>
 static void checkDimensionsAndSetDiagnostics(Sema &S, FunctionDecl *New,
                                              FunctionDecl *Old) {
-  AttributeType *NewDeclAttr = New->getAttr<AttributeType>();
-  AttributeType *OldDeclAttr = Old->getAttr<AttributeType>();
+  const auto *NewDeclAttr = New->getAttr<AttributeType>();
```

Comment:
Do we still need the const auto change?

Reply:
No, but the 'const' is definitely a GOOD change (const-all-the-things!), and 'auto' is allowed/required by coding standard.